airflow/www_rbac/views.py

```diff
@@ -0,0 +1,2070 @@
+    @has_access
+    @action_logging
+    def refresh_all(self):
+        dagbag.collect_dags(only_if_updated=False)
+        flash("All DAGs are now up to date")
+        return redirect('/')
+
+    @expose('/gantt')
+    @has_access
+    @action_logging
+    def gantt(self):
+        session = settings.Session()
+        dag_id = request.args.get('dag_id')
+        dag = dagbag.get_dag(dag_id)
+        demo_mode = conf.getboolean('webserver', 'demo_mode')
+
+        root = request.args.get('root')
+        if root:
+            dag = dag.sub_dag(
+                task_regex=root,
+                include_upstream=True,
+                include_downstream=False)
+
+        dttm = request.args.get('execution_date')
+        if dttm:
+            dttm = pendulum.parse(dttm)
+        else:
+            dttm = dag.latest_execution_date or timezone.utcnow()
+
+        form = DateTimeForm(data={'execution_date': dttm})
+
+        tis = [
+            ti for ti in dag.get_task_instances(session, dttm, dttm)
+            if ti.start_date]
+        tis = sorted(tis, key=lambda ti: ti.start_date)
+
+        tasks = []
+        for ti in tis:
+            end_date = ti.end_date if ti.end_date else timezone.utcnow()
+            tasks.append({
+                'startDate': wwwutils.epoch(ti.start_date),
+                'endDate': wwwutils.epoch(end_date),
+                'isoStart': ti.start_date.isoformat()[:-4],
+                'isoEnd': end_date.isoformat()[:-4],
+                'taskName': ti.task_id,
+                'duration': "{}".format(end_date - ti.start_date)[:-4],
+                'status': ti.state,
+                'executionDate': ti.execution_date.isoformat(),
+            })
+        states = {ti.state: ti.state for ti in tis}
+        data = {
+            'taskNames': [ti.task_id for ti in tis],
+            'tasks': tasks,
+            'taskStatus': states,
+            'height': len(tis) * 25 + 25,
+        }
+
+        session.commit()
+        session.close()
+
+        return self.render(
+            'airflow/gantt.html',
+            dag=dag,
+            execution_date=dttm.isoformat(),
+            form=form,
+            data=json.dumps(data, indent=2),
+            base_date='',
+            demo_mode=demo_mode,
+            root=root,
+        )
+
+    @expose('/object/task_instances')
+    @has_access
+    @action_logging
+    def task_instances(self):
+        session = settings.Session()
+        dag_id = request.args.get('dag_id')
+        dag = dagbag.get_dag(dag_id)
+
+        dttm = request.args.get('execution_date')
+        if dttm:
+            dttm = pendulum.parse(dttm)
+        else:
+            return ("Error: Invalid execution_date")
+
+        task_instances = {
+            ti.task_id: alchemy_to_dict(ti)
+            for ti in dag.get_task_instances(session, dttm, dttm)}
+
+        return json.dumps(task_instances)
+
+    @expose('/variables/<form>', methods=["GET", "POST"])
+    @has_access
```

Comment:
I'm not too familiar with FAB yet, but is this action protected/restricted to a different ACL to other more generic views?

Reply:
ACLs are explicitly managed in FAB (see security.py), the only thing @has_access does is to say this is a secured method. 
Generic views (the model views) under the hood uses the same mechanism (they have @has_access annotaitons on the add/edit/list methods).

Reply:
I guess by generic I actually meant "less dangerous"/read only. i.e. is getting or editing variables a different permission you can assign to users/groups from say "viewing the list of dags" or any of the other actions.

Reply:
@ashb  np, the decoupling of permissions (and roles) from actions is one of my favorite property of FAB's security model :)